lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from "path";
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([filePath, digest]) => {
+      return {
+        location: filePath,
+        digest,
+      };
+    },
+  );
+
+  const resultAggregatedByPath = jarFingerprints.reduce(
+    (accumulator, current) => {
+      const location = path.dirname(current.location);
+      accumulator[location] = accumulator[location] || [];
```

Comment:
nit: could write to `accumulator[location]` only the first time, when the array is not initialized and needs to be created